pandas/tools/merge.py

```diff
@@ -1626,7 +1626,11 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
                 clean_objs.append(v)
             objs = clean_objs
             name = getattr(keys, 'name', None)
-            keys = Index(clean_keys, name=name)
+            # GH 14848
```

Comment:
@jreback It's not a grouper issue.  Here's code that fails with 0.19.2 when passing an `Index` of tuples into `concat`.
```
df1 = pd.DataFrame({"x" : [1,2,3], "k" : [(0,0,1),(0,1,0),(1,0,0)]}).set_index('k')
df2 = pd.DataFrame({"x" : [4,5,6]}, index = pd.Index(['a','b','c']))
ic2 = df1.index
ic1 = pd.Index(['g','h','i'], name = 'foobar')
c1 = pd.concat([df2, df2, df2], axis = 0, keys=ic1, levels=[ic1], names=[ic1.name])
print('c1 is ok')
c2 = pd.concat([df2, df2, df2], axis = 0, keys=ic2, levels=[ic2], names=[ic2.name])
print('c2 is ok')
```
The value of `c1` is computed fine.  The value of `c2` raises an error because the tuples were passed as the index into `concat`.  But if you use the code that I put in above in `merge.py`, then this example works.

I know the example is a bit far-fetched, but it is equivalent to what is going on when the grouper calls `concat`, and I'd rather not mess with the grouper when the simple fix handles this particular bug.